From 0938c8ef3138421e4951542d3974a510a576e2e8 Mon Sep 17 00:00:00 2001
Message-Id: <0938c8ef3138421e4951542d3974a510a576e2e8.1422661123.git.feitong.yi@intel.com>
In-Reply-To: <655f3b24c05e6bd5aba8a8b208743a6290e46fef.1422661123.git.feitong.yi@intel.com>
References: <655f3b24c05e6bd5aba8a8b208743a6290e46fef.1422661123.git.feitong.yi@intel.com>
From: Tim Gore <tim.gore@intel.com>
Date: Wed, 7 Jan 2015 13:14:11 +0000
Subject: [PATCH 2/4] FROM_UPSTREAM [VPG]: drm/i915 Add flags check to
 i915_gem_wait_ioctl

Add flags check to i915_gem_wait_ioctl. These flags are
unused and should be zero. Somewhat trivial but there
is an igt test for this which fails.
Note that although this is from upstream, the upstream
patch could not be applied directly due to other
changes to this file

Issue: GMINL-5035
Change-Id: I72d8de8a89004a34e4f314bc1417d39f3cabdab3
Signed-off-by: Tim Gore <tim.gore@intel.com>
Upstream-repo: git://anongit.freedesktop.org/drm-intel
Upstream-commit: 11b5d5112c51f8cf4ac51f9f063bb76992a5ebe7
Upstream-as-of-tag: drm-intel-nightly
---
 drivers/gpu/drm/i915/i915_gem.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_gem.c b/drivers/gpu/drm/i915/i915_gem.c
index d8c72cd..0108f19 100644
--- a/drivers/gpu/drm/i915/i915_gem.c
+++ b/drivers/gpu/drm/i915/i915_gem.c
@@ -3067,6 +3067,9 @@ i915_gem_wait_ioctl(struct drm_device *dev, void *data, struct drm_file *file)
 	unsigned reset_counter;
 	int ret = 0;
 
+	if (args->flags != 0)
+		return -EINVAL;
+
 	if (args->timeout_ns >= 0) {
 		timeout_stack = ns_to_timespec(args->timeout_ns);
 		timeout = &timeout_stack;
-- 
1.7.9.5

